app/packages/looker/src/lookers/abstract.ts

```diff
@@ -517,9 +530,73 @@ export abstract class AbstractLooker<
   abstract updateOptions(options: Partial<State["options"]>): void;
 
   updateSample(sample: Sample) {
+    // todo: sometimes instance in spotlight?.updateItems() is defined but has no ref to sample
+    // this crashes the app. this is a bug and should be fixed
+    if (!this.sample) {
+      return;
+    }
+
+    const id = sample.id ?? sample._id;
+    const updateTimeoutMs = 10000;
+
+    if (UPDATING_SAMPLES_IDS.has(id)) {
+      UPDATING_SAMPLES_IDS.delete(id);
+      this.cleanOverlays(true);
+
+      // to prevent deadlock, we'll remove the id from the set after a timeout
+      const timeoutId = setTimeout(() => {
+        UPDATING_SAMPLES_IDS.delete(id);
+      }, updateTimeoutMs);
```

Comment:
I don't see how this has an effect, noting that id is deleted on 542, as well.

Can UPDATING_SAMPLES_IDS be replace with state internal to Looker? Like isBatching? Introducing a global store feels somewhat inflexible